src/common/snippets/src/lowered/pass/brgemm_blocking.cpp

```diff
@@ -98,7 +98,11 @@ size_t BrgemmBlockingBase::get_default_n_blk(size_t n) const {
     return 64;
 }
 size_t BrgemmBlockingBase::get_default_k_blk(size_t k) const {
+#if defined(OPENVINO_ARCH_ARM64)
+    return k;
+#else
     return !utils::is_dynamic_value(k) && k > 1024 ? 1024 : 512;
+#endif
```

Comment:
Common part of `Snippets` should not have any HW dependencies using `ifdef`

If any backend requires the specific block size for `K` dimension - this backend should provide own pass which sets needed information. I see that you implemented `GemmCPUBlocking` pass for ARM64. The method `get_default_k_blk` is virtual - you should just override it in your pass.